JwtAuthMinimalApi/Controllers/AuthController.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using JwtAuthMinimalApi.Services;
+using JwtAuthMinimalApi.Models;
+
+namespace JwtAuthMinimalApi.Controllers
+{
+    /// <summary>
+    /// Controller responsible for handling authentication-related endpoints.
+    /// </summary>
+    [ApiController]
+    [Route("api/auth")]
+    public class AuthController : ControllerBase
+    {
+        private readonly IJwtTokenGenerator _jwtTokenGenerator;
+        private readonly IAuthorizationService _authorizationService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthController"/> class.
+        /// </summary>
+        /// <param name="jwtTokenGenerator">Service to generate JWT tokens.</param>
+        public AuthController(IJwtTokenGenerator jwtTokenGenerator, IAuthorizationService authorizationService)
+        {
+            _jwtTokenGenerator = jwtTokenGenerator;
+            _authorizationService = authorizationService;
+        }
+
+        /// <summary>
+        /// Authenticates the user with provided credentials and returns a JWT token if successful.
+        /// </summary>
+        /// <param name="request">The login request containing username and password.</param>
+        /// <returns>
+        /// 200 OK with JWT token on successful authentication,  
+        /// 400 Bad Request if username or password is missing,  
+        /// 401 Unauthorized if credentials are invalid.
+        /// </returns>
+        [HttpPost("login")]
+        public IActionResult Login([FromBody] LoginRequest request)
+        {
+            if (string.IsNullOrEmpty(request.Username) || string.IsNullOrEmpty(request.Password))
+            {
+                return BadRequest("Username and password are required");
+            }
+
+            if (_authorizationService.ValidateCredentials(request.Username, request.Password))
+            {
+                var token = _jwtTokenGenerator.GenerateToken(request.Username);
```

Comment:
the auth service should be responsible to authorize and validate the login and return success with the token or failure. All this logic belongs to the auth service not the controller.